errors/errors.go

```diff
@@ -46,128 +46,126 @@ func GetDetails(err error) map[string]string {
-	ErrBadRange                       = errors.New("storage: bad range")
-	ErrBadLayerCount                  = errors.New("manifest: layers count doesn't correspond to config history")
-	ErrManifestConflict               = errors.New("manifest: multiple manifests found")
-	ErrManifestMetaNotFound           = errors.New("metadb: image metadata not found for given manifest reference")
-	ErrImageMetaNotFound              = errors.New("metadb: image meta not found")
-	ErrUnexpectedMediaType            = errors.New("metadb: got unexpected media type")
-	ErrRepoMetaNotFound               = errors.New("metadb: repo metadata not found for given repo name")
-	ErrTagMetaNotFound                = errors.New("metadb: tag metadata not found for given repo and tag names")
-	ErrTypeAssertionFailed            = errors.New("storage: failed DatabaseDriver type assertion")
-	ErrInvalidRequestParams           = errors.New("resolver: parameter sent has invalid value")
+	ErrBadConfig                      = errors.New("invalid server config")
+	ErrCliBadConfig                   = errors.New("invalid cli config")
+	ErrRepoNotFound                   = errors.New("repository not found")
+	ErrRepoBadVersion                 = errors.New("repository unsupported layout version")
+	ErrRepoBadLayout                  = errors.New("repository invalid layout")
+	ErrManifestNotFound               = errors.New("manifest not found")
+	ErrBadManifest                    = errors.New("manifest invalid contents")
+	ErrUploadNotFound                 = errors.New("uploads not found")
+	ErrBadUploadRange                 = errors.New("uploads bad range")
+	ErrBlobNotFound                   = errors.New("blob not found")
+	ErrBadBlob                        = errors.New("bad blob")
+	ErrBadBlobDigest                  = errors.New("bad blob digest")
+	ErrBlobReferenced                 = errors.New("blob referenced by manifest")
+	ErrManifestReferenced             = errors.New("manifest referenced by index image")
+	ErrUnknownCode                    = errors.New("unknown error code")
+	ErrBadCACert                      = errors.New("invalid tls ca cert")
+	ErrBadUser                        = errors.New("non-existent user")
+	ErrEntriesExceeded                = errors.New("too many entries returned")
+	ErrLDAPEmptyPassphrase            = errors.New("empty ldap passphrase")
+	ErrLDAPBadConn                    = errors.New("bad ldap connection")
+	ErrLDAPConfig                     = errors.New("invalid LDAP configuration")
+	ErrCacheRootBucket                = errors.New("unable to create/update root cache bucket")
+	ErrCacheNoBucket                  = errors.New("unable to find cache bucket")
+	ErrCacheMiss                      = errors.New("cache miss")
+	ErrRequireCred                    = errors.New("bind ldap credentials required")
+	ErrInvalidCred                    = errors.New("invalid ldap credentials")
+	ErrEmptyJSON                      = errors.New("cli config json is empty")
```

Comment:
How is this different from ErrCliBadConfig? Why not use that?

Reply:
I think it's more specific, maybe we should return a merge of ErrCliBadConfig with ErrEmptyJSON

Reply:
I thought we wanted the errors a bit more generic?